app/src/main/java/dev/baseio/discordjetpackcompose/ui/routes/dashboard/friends/components/FriendRowComponent.kt

```diff
@@ -0,0 +1,179 @@
+                )
+                Spacer(modifier = Modifier.width(30.dp))
+                ActionButton(
+                    painterResource =
+                    R.drawable.ic_baseline_person_add_alt_1_24,
+                    contentDescription = stringResource(id = R.string.add_friend)
+                )
+            }
+        }
+    }
+}
+
+@Composable
+fun ActionButton(
+    background: Color = discord_icon_button_bg,
+    icon: ImageVector? = null,
+    painterResource: Int? = null,
+    contentDescription: String,
+    tintColor: Color = icon_button
+) {
+    IconButton(
+        onClick = { }, modifier = Modifier
+            .size(25.dp)
+            .clip(CircleShape)
+            .background(background)
+    ) {
+        if (icon != null) {
+            Icon(
+                icon,
+                contentDescription = contentDescription,
+                tint = tintColor,
+                modifier = Modifier.size(20.dp)
+            )
+        } else if (painterResource != null) {
+            Icon(
+                painter = painterResource(id = painterResource),
+                contentDescription = contentDescription,
+                tint = tintColor,
+                modifier = Modifier.size(20.dp)
+            )
+        }
+    }
+}
+
+@Composable
+fun ProfileComponent(chatUserEntity: ChatUserEntity, isFriend: Boolean = true) {
+    Row(
+        modifier = Modifier.wrapContentWidth(),
+        horizontalArrangement = Arrangement.Start,
+        verticalAlignment = Alignment.CenterVertically
+    ) {
+        Box(
+            contentAlignment = Alignment.BottomEnd,
+            modifier = Modifier.padding(8.dp),
+        ) {
+            AsyncImage(
+                model = ImageRequest.Builder(LocalContext.current)
+                    .data(chatUserEntity.profileImage)
+                    .placeholder(R.drawable.light_app_logo)
+                    .crossfade(true)
+                    .build(),
+                contentDescription = null,
+                contentScale = ContentScale.Crop,
+                modifier = Modifier
+                    .width(32.dp)
+                    .aspectRatio(1f)
+                    .clip(CircleShape),
+            )
+            if (isFriend) {
+                OnlineIndicator(isOnline = chatUserEntity.isOnline)
+            }
+        }
+        Column(
+            modifier = Modifier.padding(start = 8.dp),
+        ) {
+            Text(
+                text = chatUserEntity.username, style = DirectMessageListTypography.h6.copy(
+                    fontWeight = if (isFriend) {
+                        FontWeight.Normal
+                    } else {
+                        FontWeight.Bold
+                    }
+                ),
+                maxLines = 1,
+                overflow = TextOverflow.Ellipsis,
+            )
+            if (isFriend) {
+                Text(
+                    text = chatUserEntity.currentStatus?:"",
```

Comment:
(Optional) There's a convenience method called `.orEmpty()`. We can use that instead of writing `?: ""`